Mage.Tests/src/test/java/org/mage/test/cards/rules/DontUntapTest.java

```diff
@@ -73,5 +73,127 @@ public void TestAjaniVengeantFirst() {
+        addCard(Zone.GRAVEYARD, playerA, "Reassembling Skeleton", 1);
+
+        addCard(Zone.BATTLEFIELD, playerB, "Island", 3);
+        addCard(Zone.HAND, playerB, "Exhaustion", 1);
+        
+        attack(1, playerA, "Silvercoat Lion");
+        castSpell(2, PhaseStep.PRECOMBAT_MAIN, playerB, "Exhaustion", playerA);
+        activateAbility(2, PhaseStep.END_TURN, playerA, "{1}{B}:"); //Reassembling Skeleton enters after spell is cast, should remain tapped
+
+        setStrictChooseMode(true);
+        setStopAt(3, PhaseStep.PRECOMBAT_MAIN);
+        execute();
+
+        assertTapped("Silvercoat Lion", true);
+        assertTapped("Reassembling Skeleton", true);
+        assertTapped("Swamp", true);
+    }
+
+    @Test
+    public void testManaVaporsEffect() {
+        addCard(Zone.BATTLEFIELD, playerA, "Mountain", 1);
+        addCard(Zone.HAND, playerA, "Lightning Bolt", 1);
+
+        addCard(Zone.BATTLEFIELD, playerB, "Island", 2);
+        addCard(Zone.HAND, playerB, "Mana Vapors", 1);
+        
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Lightning Bolt", playerB);
+        castSpell(2, PhaseStep.PRECOMBAT_MAIN, playerB, "Mana Vapors", playerA);
+
+        setStrictChooseMode(true);
+        setStopAt(3, PhaseStep.PRECOMBAT_MAIN);
+        execute();
+
+        assertTapped("Mountain", true);
+    }
+
+    @Test
+    public void testMisstepEffect() {
+        addCard(Zone.BATTLEFIELD, playerA, "Silvercoat Lion", 1);
+        addCard(Zone.BATTLEFIELD, playerA, "Swamp", 2);
+        addCard(Zone.GRAVEYARD, playerA, "Reassembling Skeleton", 1);
+
+        addCard(Zone.BATTLEFIELD, playerB, "Island", 3);
+        addCard(Zone.HAND, playerB, "Exhaustion", 1);
         
+        attack(1, playerA, "Silvercoat Lion");
+        castSpell(2, PhaseStep.PRECOMBAT_MAIN, playerB, "Exhaustion", playerA);
+        activateAbility(2, PhaseStep.END_TURN, playerA, "{1}{B}:"); //Reassembling Skeleton enters after spell is cast, should remain tapped
+
+        setStrictChooseMode(true);
+        setStopAt(3, PhaseStep.PRECOMBAT_MAIN);
+        execute();
+
+        assertTapped("Silvercoat Lion", true);
+        assertTapped("Reassembling Skeleton", true);
+    }
+
+    @Test
+    public void testPollenLullabyEffect() {
+        addCard(Zone.BATTLEFIELD, playerA, "Silvercoat Lion", 1);
+        addCard(Zone.BATTLEFIELD, playerA, "Swamp", 2);
+        addCard(Zone.GRAVEYARD, playerA, "Reassembling Skeleton", 1);
+
+        addCard(Zone.BATTLEFIELD, playerB, "Plains", 2);
+        addCard(Zone.HAND, playerB, "Pollen Lullaby", 1);
+
+        removeAllCardsFromLibrary(playerA);
+        removeAllCardsFromLibrary(playerB);
+        addCard(Zone.LIBRARY, playerA, "Island", 8);
+        addCard(Zone.LIBRARY, playerB, "Silvercoat Lion", 8);
+        skipInitShuffling();
+
+        attack(1, playerA, "Silvercoat Lion");
+        castSpell(1, PhaseStep.DECLARE_BLOCKERS, playerB, "Pollen Lullaby");
+        setChoice(playerB, "PlayerA");
```

Comment:
Good practice is to include an inline comment explaining what each choice is, and/or include a multiline comment with the oracle text of the card being tested at the top.

(Also, could just use the `playerA` variable here.)

Reply:
Not sure what you mean by using the `playerA` variable, if I try to put it in the choice it gives me an error, and if I put it in castSpell, the test fails because the spell doesn't specify a target. I think it might be some weirdness with Clash.

Reply:
Alright, disregard that part then